source/slang/hlsl.meta.slang

```diff
@@ -199,13 +199,15 @@ struct StructuredBuffer
         out uint numStructs,
         out uint stride);
 
-    __target_intrinsic(glsl, "$0._data[$1]") 
+    __target_intrinsic(glsl, "$0._data[$1]")
+    __target_intrinsic(spirv_direct, "%addr = 65 resultType*StorageBuffer resultId _0 _1; 61 resultType resultId %addr;")
```

Comment:
I like the human-readable syntax here. I haven't looked at the implementation yet, but I can already guess from what is written here how most of it works.

My initial notes, upon looking at this:

* It might be sensible to have something like `_r` instead of `resultId`, to communicate that it is similar to `_0`, `_1`, etc.

* It might be sensible to have a uniform way to refer to the type of an operand or result, like `_t0` for the type of operand 0, and `_tr` for the type of the result.

* It might eventually make sense to allow simple identifiers for the opcode names instead of magic numbers, even if doing so requires us to have a string->id mapping table somewhere. This could always be added later, though, since we'd want to support hard-coded numbers as well.

* The `resultType*StorageBuffer` bit is probably the messiest thing there, obviously. It might be nice to have the SPIR-V "assembly" format being used here support a simple but uniform form of Lisp-y nested expressions, so that `(Ptr resultType StorageBuffer)` would be valid. Allowing nested expressions would seemingly allow this example to be written as:

```
61 resultType resultId (65 (Ptr resultType StorageBuffer) resultId _0 _1)
```

The implied rules here would be that a `(...)`-enclosed range represents another SPIR-V instruction that will be emitted into the flow. The logical evaluation order would be left-to-right depth-first (no ambiguity). For any nested instruction with an opcode that represents a type, etc. it would go through the existing deduplication we'd have to do for SPIR-V types anyway, and would be emitted into the relevant section (instead of directly into a function body).

Again, none of these "need" to happen, as there are no actual problems with what's here now.